### PR TITLE
Update changes from development

### PR DESCRIPTION
**Pull request by pugakn**
- High priority Fix for the voting Unity on proposals.

### CODE REVIEW

File: src/components/profiles/proposal-item.vue [hunk 1]
```diff
@@ -202,12 +202,13 @@ export default {
     // TODO: Move this to a mixin
     calculateVoting (proposal) {
       if (proposal && proposal.votetally && proposal.votetally.length) {
-        const passCount = parseFloat(proposal.pass.count)
-        const failCount = parseFloat(proposal.fail.count)
+        // const passCount = parseFloat(proposal.pass.count)
+        // const failCount = parseFloat(proposal.fail.count)
         const abstain = parseFloat(proposal.votetally[0].abstain_votePower_a)
```
Comment (n13) [inline src/components/profiles/proposal-item.vue:207]: why is this commented code here? do we need this?

File: src/components/profiles/proposal-item.vue [hunk 2]
```diff
@@ -202,12 +202,13 @@ export default {
...
         const pass = parseFloat(proposal.votetally[0].pass_votePower_a)
         const fail = parseFloat(proposal.votetally[0].fail_votePower_a)
-        const unity = (passCount + failCount > 0) ? passCount / (passCount + failCount) : 0
+
+        const unity = (pass + fail > 0) ? pass / (pass + fail) : 0
         let supply = this.supply
```
Comment (n13) [inline src/components/profiles/proposal-item.vue:212]: looks good

File: src/components/proposals/proposal-list.vue [hunk 3]
```diff
@@ -102,12 +102,13 @@ export default {
...
-        const failCount = parseFloat(proposal.fail.count)
+        // const passCount = parseFloat(proposal.pass.count)
+        // const failCount = parseFloat(proposal.fail.count)
         const abstain = parseFloat(proposal.votetally[0].abstain_votePower_a)
         const pass = parseFloat(proposal.votetally[0].pass_votePower_a)
         const fail = parseFloat(proposal.votetally[0].fail_votePower_a)
-        const unity = (passCount + failCount > 0) ? passCount / (passCount + failCount) : 0
+
+        const unity = (pass + fail > 0) ? pass / (pass + fail) : 0
         let supply = this.supply
```
Comment (n13) [inline src/components/proposals/proposal-list.vue:112]: maybe for now OK but this duplicated code needs to be unified somewhere in a central location.

Comment (n13) [reply src/components/proposals/proposal-list.vue:112]: Is it possible to quickly move this code to one location

We can't have the same calculation in different areas of the code, breaks the DRY rule (don't repeat yourself)

File: src/components/proposals/voting.vue [hunk 4]
```diff
@@ -342,7 +346,7 @@ widget(:title="widgetTitle" noPadding :background="background" :textColor="expir
         q-btn.q-px-xl.full-width(v-if="(expired || vote ) && !approved" no-caps rounded color="white" outline :class="{ 'no-pointer-events': expired, ...backgroundButton }" :disable="proposed && expired" @click="voting = !voting") {{ voteString }}
          q-tooltip You can change your vote
         q-btn.q-mt-xs.full-width(v-if="proposed && active && accepted && expired" unelevated no-caps rounded color="white" text-color="positive" @click="onActive") Activate
-        q-btn.q-mt-xs.full-width(v-if="expired && !accepted && active && !archived" unelevated no-caps rounded color="white" text-color="negative" @click="onActive") Archive
+        q-btn.q-mt-xs.full-width(v-if="expired && !accepted && active && !rejected" unelevated no-caps rounded color="white" text-color="negative" @click="onActive") Archive
```
Comment (n13) [inline src/components/proposals/voting.vue:349]: was rejected mixed up with archived?

File: src/mixins/validation.js [hunk 5]
```diff
@@ -22,6 +22,7 @@ export const validation = {
         maxLength: val => value => value.length <= val || `This field must contain less than ${val} characters`,
         isAccountAvailable: async account => (await this.isAccountFree(account.toLowerCase())) || `The account "${account}" already exists`,
         accountExists: async account => !(await this.isAccountFree(account.toLowerCase())) || `The account "${account}" doesn't exist`,
+        isTokenAvailable: async token => (await this.isTokenFree(token.toUpperCase())) || `The token "${token}" already exists`,
```
Comment (n13) [inline src/mixins/validation.js:25]: this change seems unrelated to the hotfix - it's a different fix.

**Review by n13 — CHANGES_REQUESTED** [review]

File: src/components/proposals/proposal-list.vue (hunk 3 above)
Comment (pugakn) [reply src/components/proposals/proposal-list.vue:110]: Will be done later

File: src/components/proposals/voting.vue (hunk 4 above)
Comment (pugakn) [reply src/components/proposals/voting.vue:349]: Yeah, there was a bug where we were showing archived instead rejected state

**Review by n13 — APPROVED** [review]
need hotfix so for this time we allow dev changes in
